lib/proxy.js

```diff
@@ -174,7 +174,7 @@ function ReverseProxy(opts){
     if(target){
       proxy.ws(req, socket, head, {target: target});
     }else{
-      notFound(socket);
+      respondNotFound(req, socket);
```

Comment:
:P
If we don't seem to really care about this use case, we can perhaps open a separate issue and merge this? I haven't actually tested whether sockets work at all, so I don't know haha